packages/itwin/tree-widget/src/components/trees/models-tree/ModelsTreeComponent.tsx

```diff
@@ -124,13 +125,20 @@ function ModelsTreeComponentImpl(props: ModelTreeComponentProps & { iModel: IMod
   const contentClassName = classNames("tree-widget-tree-content", props.density === "enlarged" && "enlarge");
 
   useEffect(() => {
-    queryModelsForHeaderActions(iModel)
-      .then((modelInfos: ModelInfo[]) => {
-        setAvailableModels(modelInfos);
-      })
-      .catch((_e) => {
-        setAvailableModels([]);
-      });
+    const queryModels = () => {
+      queryModelsForHeaderActions(iModel)
+        .then((modelInfos: ModelInfo[]) => {
+          setAvailableModels(modelInfos);
+        })
+        .catch((_e) => {
+          setAvailableModels([]);
+        });
+    };
+    queryModels();
```

Comment:
Probably need to do something to avoid race conditions?

Reply:
You are right, updated.